src/runtime_src/core/include/experimental/xrt_pscontext.h

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#ifndef XRT_PSCONTEXT_H_
+#define XRT_PSCONTEXT_H_
+
+#include "xrt/detail/config.h"
+#include "xrt/detail/pimpl.h"
+#include "xclhal2.h"
+
+#ifdef __cplusplus
+#include <memory>
+
+/*
+ * PS Context Data Structure to be derived from for user's xrtHandle class
+ * User will need to declare an xrtHandles class derived from xrt::pscontext class
+ * Example:
+ * class xrtHandles : public xrt::pscontext
+ * {
+ * public:
+ *   xrt::device dhdl;
+ *   xrt::kernel kernel;
+ *   xrtHandles(xclDeviceHandle dhdl_in, const xuid_t xclbin_uuid)
```

Comment:
Do not use `xclDeviceHandle`  use `xrt::device`.
And use xrt::uuid instead of plain uuid_t (aka xuid_t).